docs/sql-ref-functions-builtin-window.md

```diff
@@ -0,0 +1,215 @@
+
+<table class="table">
+  <thead>
+    <tr><th style="width:25%">Function</th><th>Description</th></tr>
+  </thead>
+    <tr>
+      <td><b> cume_dist </b></td>
+      <td>Returns the cumulative distribution of values within a window partition, i.e. the fraction of rows that are below the current row. This is equivalent to the CUMEDIST function in SQL. </td>
+    </tr>
+    <tr>
+      <td><b> lag </b></td>
+      <td>Returns the value that is <code>offset</code> rows before the current row, and <code>null</code> if there are less than <code>offset</code> rows before the current row.</td>
+    </tr>
+    <tr>
+      <td><b> lead </b></td>
+      <td>Returns the value that is <code>offset</code> rows after the current row, and <code>null</code> if there are less than <code>offset</code> rows after the current row. This is equivalent to the LEAD function in SQL.</td>
+    </tr>
+</table>
+
+#### Aggregate Functions
+
+Any of the aggregate functions can be used as a window function. Please refer to the complete list of Spark [Aggregate Functions](sql-ref-functions-builtin-aggregate.html).
+
+### How to Use Window Functions
+
+  * Mark a function as window function by using `over`.
+    - SQL: Add an OVER clause after the window function, e.g. avg (...) OVER (...);
+    - DataFrame API: Call the window function's `over` method, e.g. rank().over(...)
+  * Define the window specification associated with this function. A window specification includes partitioning specification, ordering specification, and frame specification.
+    - Partitioning Specification:
+      - SQL: PARTITION BY
+      - DataFrame API: Window.partitionBy (...)
+    - Ordering Specification:
+      - SQL: Order BY
+      - DataFrame API: Window.orderBy (...)
+    - Frame Specification:
+      - SQL: ROWS (for ROW frame), RANGE (for RANGE frame)
+      - DataFrame API: WindowSpec.rowsBetween (for ROW frame), WindowSpec.rangeBetween (for RANGE frame)
+
+### Examples
+
+{% highlight scala %}
+
+  import spark.implicits._
+
+  val data = Seq(("Lisa", "Sales", 10000),
+    ("Evan", "Sales", 32000),
+    ("Fred", "Engineering", 21000),
+    ("Helen", "Marketing", 29000),
+    ("Alex", "Sales", 30000),
+    ("Tom", "Engineering", 23000),
+    ("Jane", "Marketing", 29000),
+    ("Jeff", "Marketing", 35000),
+    ("Paul", "Engineering", 29000),
+    ("Chloe", "Engineering", 23000)
+  )
+  val df = data.toDF("name", "dept", "salary")
+  df.show()
+  // +-----+-----------+------+
+  // | name|       dept|salary|
+  // +-----+-----------+------+
+  // | Lisa|      Sales| 10000|
+  // | Evan|      Sales| 32000|
+  // | Fred|Engineering| 21000|
+  // |Helen|  Marketing| 29000|
+  // | Alex|      Sales| 30000|
+  // |  Tom|Engineering| 23000|
+  // | Jane|  Marketing| 29000|
+  // | Jeff|  Marketing| 35000|
+  // | Paul|Engineering| 29000|
+  // |Chloe|Engineering| 23000|
+  // +-----+-----------+------+
+
+  val windowSpec = Window.partitionBy("dept").orderBy("salary")
```

Comment:
We don't need to write this document in the same SQL way with https://github.com/apache/spark/pull/28120?

Reply:
I will  make it consistent with Kevin's PR. I really want to keep the function tables simple, though, because the user can refer to API docs for details. Are you OK with the functions tables, or you want me to change those too?

Reply:
Ah, I see. I personally think its better to follow the SQL format here, too.

Reply:
![image](https://user-images.githubusercontent.com/13592258/78866173-c063c800-79f3-11ea-8b96-f07cc165490d.png)

We are talking about this table, right? :)

Reply:
oh, my bad. Yea, the consistent format is better, so I personally think its better to follow the Kevin's format.

Reply:
Probably, users who wanna see simpler built-in function docs can check 
https://spark.apache.org/docs/3.0.0-preview/api/sql/index.html. So, I think it is better to write this page as detailed as possible.